astroquery/mast/missions.py

```diff
@@ -237,25 +275,29 @@ def get_column_list(self):
 
         Returns
         -------
-        response : `~astropy.table.Table` that contains columns names, types  and their descriptions
+        response : `~astropy.table.Table` that contains columns names, types, and descriptions
         """
 
-        url = f"{conf.server}/search/util/api/v0.1/column_list?mission={self.mission}"
-
-        try:
-            results = requests.get(url)
-            results = results.json()
-            rows = []
-            for result in results:
-                result.pop('field_name')
-                result.pop('queryable')
-                result.pop('indexed')
-                result.pop('default_output')
-                rows.append((result['column_name'], result['qual_type'], result['description']))
-            data_table = Table(rows=rows, names=('name', 'data_type', 'description'))
-            return data_table
-        except Exception:
-            raise Exception(f"Error occurred while trying to get column list for mission {self.mission}")
+        if not self.columns.get(self.mission):
+            try:
+                # Send server request to get column list for current mission
+                params = {'mission': self.mission}
+                resp = utils._simple_request(f'{conf.server}/search/util/api/v0.1/column_list', params)
+
+                # Parse JSON and extract necessary info
+                results = resp.json()
+                rows = [
+                    (result['column_name'], result['qual_type'], result['description'])
+                    for result in results
+                ]
+
+                # Create Table with parsed data
+                col_table = Table(rows=rows, names=('name', 'data_type', 'description'))
+                self.columns[self.mission] = col_table
+            except Exception:
```

Comment:
I see that this is inherited code, but we should be more specific and only except specific exceptions rather than staying this super general. And if we need to stay this general, capture the exception and raise the same more specific type of it rather than reraise the generic `Exception`

Reply:
Modified to catch more specific exceptions